src/app/ReadClient.cpp

```diff
@@ -48,14 +88,31 @@ ReadClient::ReadClient(InteractionModelEngine * apImEngine, Messaging::ExchangeM
     }
 }
 
+void ReadClient::ClearActiveSubscriptionState()
+{
+    mIsInitialReport           = true;
+    mIsPrimingReports          = true;
+    mPendingMoreChunks         = false;
+    mMinIntervalFloorSeconds   = 0;
+    mMaxIntervalCeilingSeconds = 0;
+    mSubscriptionId            = 0;
+    MoveToState(ClientState::Idle);
+}
+
+void ReadClient::StopResubscription()
+{
+    ClearActiveSubscriptionState();
+    CancelLivenessCheckTimer();
+    CancelResubscribeTimer();
+    mpCallback.OnDeallocatePaths(std::move(mReadPrepareParams));
+}
+
 ReadClient::~ReadClient()
 {
     Abort();
 
     if (IsSubscriptionType())
     {
-        CancelLivenessCheckTimer();
```

Comment:
Why is this OK to remove?  What canceled this timer if we are a subscription that is _not_ doing resubscriptions?

Reply:
good catch, accidental removal, fixed it, thanks

Reply:
fixed, thanks